packages/cli/src/fulltext/FileIndex.ts

```diff
@@ -298,3 +304,20 @@ export async function filterFiles(
   }
   return result;
 }
+
+// Reads git attribute patterns from a .gitattributes file.
+export function getGitAttributes(directory: string) {
+  const gitAttributesPath = join(directory, '.gitattributes');
+  if (!existsSync(gitAttributesPath)) return [];
+
+  const gitAttributesContent = readFileSync(gitAttributesPath, 'utf-8');
```

Comment:
[From node.js doc](https://nodejs.org/api/fs.html#fsexistspath-callback):

> Using fs.exists() to check for the existence of a file before calling fs.open(), fs.readFile(), or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist.

(Essentially, exists returning true doesn't guarantee that read will succeed, so you have to handle the error anyway, so might as well skip the exists.)